### PR TITLE
Add TMUS entry/logo to wall of users

### PR DESCRIPTION
**Pull request by phenixblue**
Please provide a description for what this PR is for.

Adds T-Mobile US, Inc. to the wall of users along with the company logo.

And to help us figure out who should review this PR, please 
put an X in all the areas that this PR affects.

- [ ] Configuration Infrastructure
- [X] Docs
- [ ] Installation
- [ ] Networking
- [ ] Performance and Scalability
- [ ] Policies and Telemetry
- [ ] Security
- [ ] Test and Release
- [ ] User Experience
- [ ] Developer Infrastructure

### CODE REVIEW

File: data/companies.yml [hunk 1]
```diff
@@ -472,6 +472,9 @@ users:
   - name: "US Air Force"
     logo: "/logos/usaf.svg"
     url: "https://www.af.mil/"
+  - name: "T-Mobile US"
```
Comment (ericvn) [inline data/companies.yml:475]: This should be added alphabetically within the section.

Comment (ericvn) [reply data/companies.yml:475]: Just an FYI, but you can look at https://deploy-preview-10777--preliminary-istio.netlify.app/latest/about/case-studies (or click on Details to the right of the deploy/netlify test and go through the site). You can see what will be deployed, and if you scroll down you will see T-mobile between the US Air Force and Walmart.

Comment (phenixblue) [reply data/companies.yml]: Derp....sorry about that. Pushed a commit to fix the ordering

Comment (craigbox) [reply data/companies.yml]: Hi Joe,

T-Mobile is already listed, just further up the list in the Case Studies section, highlighting your Istiocon talk from last year.

The wall of logos is "and all these others". 

Would you prefer we updated the logo to the one you've provided, as I see they are different?

Comment (phenixblue) [reply data/companies.yml]: Oh, I had not noticed it was up above. Thank you for catching that. I'll just close this PR. Sorry for missing that.

**Comment by phenixblue** [comment]
Closing since TMUS is already listed in the Case Studies section 🤦‍♂️

**Pull request closed without merging**